neuralmonkey/decoders/autoregressive.py

```diff
@@ -189,9 +193,12 @@ def embedding_matrix(self) -> tf.Variable:
     def get_logits(self, state: tf.Tensor) -> tf.Tensor:
         """Project the decoder's output layer to logits over the vocabulary."""
         state = dropout(state, self.dropout_keep_prob, self.train_mode)
+        logits = tf.matmul(state, self.decoding_w) + self.decoding_b
 
-        logits = tf.matmul(state, self.decoding_w)
-        logits += self.decoding_b
+        if self.supress_unk:
+            unk_mask = tf.one_hot(
+                UNK_TOKEN_INDEX, depth=len(self.vocabulary), on_value=-1e-9)
```

Comment:
Even though this is closed already.... Why is the on_value=-1e-9 ??? Shouldn't it be -1e9 ???
Now you are just lowering the logits of UNK_TOKENs by a very small number.